packages/protocol/contracts/baklava/Freezable.sol

```diff
@@ -0,0 +1,51 @@
+pragma solidity ^0.5.3;
+
+
+contract Freezable {
+  bool public frozen;
+  address public freezer;
+
+  // onlyFreezer functions can only be called by the specified `freezer` address
+  modifier onlyFreezer() {
+    require(msg.sender == freezer);
+    _;
+  }
+
+  // onlyWhenNotFrozen functions can only be called when `frozen` is false, otherwise they are
+  // noops.
+  modifier onlyWhenNotFrozen() {
```

Comment:
Is this necessary? I would consider cutting this function and renaming `onlyWhenNotFrozenOrThrow` to `onlyWhenNotFrozen`

Reply:
Should have cleared this up - this is already looking forward to the next change, where `distributeEpochPayments` will need to be frozen. By not reverting when that function is frozen, I think we'll be able to contain this temporary change for Baklava to smart contracts.

I could see an argument for reverting being the default semantic, happy to rename this one to `onlyWhenNotFrozenNoThrow`.

WDYT?

Reply:
Are you sure we don't want to revert when we freeze epoch rewards for TGCSO? Reverting would prevent any proof-of-stake action from happening in `Finalize`, e.g. validator score changes, payments, etc.